tests/extension/oneapi_composite_device/enumerating_composite_devices.cpp

```diff
@@ -0,0 +1,157 @@
+#ifndef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_composite device extension is not supported by an "
+      "imlementation");
+#else
+
+  std::vector<sycl::device> composite_devices;
+
+  // get_composite_device may not throw
+  try {
+    composite_devices =
+        sycl::ext::oneapi::experimental::get_composite_devices();
+  } catch (sycl::exception& e) {
+    FAIL("get_composite_devices threw an exception: " << e.what());
+  }
+
+  {
+    INFO("Subsequent calls to get_composite_devices()");
+    std::vector<sycl::device> composite_devices2;
+    // get_composite_device may not throw
+    try {
+      composite_devices2 =
+          sycl::ext::oneapi::experimental::get_composite_devices();
+      INFO("Checking that subsequent call returns the same list of devices");
+      REQUIRE(composite_devices == composite_devices2);
+    } catch (sycl::exception& e) {
+      FAIL("get_composite_devices threw an exception: " << e.what());
+    }
+  }
+
+#endif
+}
+
+TEST_CASE("Test for platform::ext_oneapi_get_composite_devices",
+          "[oneapi_composite_device]") {
+#ifndef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_composite device extension is not supported by an "
+      "imlementation");
+#else
+
+  auto platform = sycl_cts::util::get_cts_object::platform();
+  std::vector<sycl::device> composite_devices;
+
+  // get_composite_device may not throw
+  try {
+    composite_devices = platform.ext_oneapi_get_composite_devices();
+  } catch (sycl::exception& e) {
+    FAIL("platform::ext_oneapi_get_composite_devices threw an exception: "
+         << e.what());
+  }
+
+  {
+    INFO("Subsequent calls to platform::ext_oneapi_get_composite_devices()");
+    std::vector<sycl::device> composite_devices2;
+    // get_composite_device may not throw
+    try {
+      composite_devices2 = platform.ext_oneapi_get_composite_devices();
+      INFO("Checking that subsequent call returns the same list of devices");
+      REQUIRE(composite_devices == composite_devices2);
+    } catch (sycl::exception& e) {
+      FAIL("platform::ext_oneapi_get_composite_devices threw an exception: "
+           << e.what());
+    }
+  }
+
+#endif
+}
+
+TEST_CASE(
+    "Test to check that composite devices are not returned as root devices",
+    "[oneapi_composite_device]") {
+#ifndef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+  SKIP(
+      "The sycl_ext_oneapi_composite device extension is not supported by an "
+      "imlementation");
+#else
+
+  auto platform = sycl_cts::util::get_cts_object::platform();
+
+  std::vector<sycl::device> root_devices = platform.get_devices();
+  for (auto device : sycl::device::get_devices()) {
+    root_devices.push_back(device);
+  }
+
+  std::vector<sycl::device> composite_devices =
+      platform.ext_oneapi_get_composite_devices();
+  for (auto device : sycl::ext::oneapi::experimental::get_composite_devices()) {
+    composite_devices.push_back(device);
+  }
```

Comment:
Similar to the above one: the free function should include all the composite devices returned by `platform.ext_oneapi_get_composite_devices`, why do we need them both?

Reply:
Applied in f8b1ca288cbb9f405080482cf9b73cd3ed38b2eb